lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java

```diff
@@ -0,0 +1,834 @@
+      classification.setDescription("Description");
+      classification.setName("Name");
+      Thread.sleep(20); // to avoid identity of modified timestamps between classification and base
+      classificationService.updateClassification(classification);
+      classificationSecondUpdate.setName("Name again");
+      classificationSecondUpdate.setDescription("Description again");
+
+      ConcurrencyException expectedException =
+          new ConcurrencyException(classificationSecondUpdate.getId());
+      assertThatThrownBy(
+              () -> classificationService.updateClassification(classificationSecondUpdate))
+          .usingRecursiveComparison()
+          .isEqualTo(expectedException);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_ThrowException_When_TryingToUpdateClassificationWithInvalidParentId()
+        throws Exception {
+      Classification classification =
+          defaultTestClassification().buildAndStore(classificationService);
+
+      classification.setParentId("NON EXISTING ID");
+
+      ClassificationNotFoundException expectedException =
+          new ClassificationNotFoundException("NON EXISTING ID");
+      assertThatThrownBy(() -> classificationService.updateClassification(classification))
+          .usingRecursiveComparison()
+          .isEqualTo(expectedException);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_ThrowException_When_TryingToUpdateClassificationWithInvalidParentKey()
+        throws Exception {
+      Classification classification =
+          defaultTestClassification().buildAndStore(classificationService);
+
+      classification.setParentKey("NON EXISTING KEY");
+
+      ClassificationNotFoundException expectedException =
+          new ClassificationNotFoundException("NON EXISTING KEY", "DOMAIN_A");
+      assertThatThrownBy(() -> classificationService.updateClassification(classification))
+          .usingRecursiveComparison()
+          .isEqualTo(expectedException);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_ThrowException_When_TryingToUpdateClassificationWithOwnKeyAsParentKey()
+        throws Exception {
+      Classification classification =
+          defaultTestClassification().buildAndStore(classificationService);
+
+      classification.setParentKey(classification.getKey());
+
+      InvalidArgumentException expectedException =
+          new InvalidArgumentException(
+              String.format(
+                  "The Classification '%s' has the same key and parent key",
+                  classification.getName()));
+      assertThatThrownBy(() -> classificationService.updateClassification(classification))
+          .usingRecursiveComparison()
+          .isEqualTo(expectedException);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_NotThrowException_When_UpdatingClassificationWithEmptyServiceLevel()
```

Comment:
What do you think about moving this test to the top class or to `UpdatePriorityAndServiceLevelTest `and checking if the serviceLevel is updated to "" correctly?